hydrolib/core/dimr/models.py

```diff
@@ -342,3 +438,73 @@ def _get_serializer(
     @classmethod
     def _get_parser(cls) -> Callable:
         return DIMRParser.parse
+
+    @classmethod
+    def _parse(cls, path: Path) -> Dict:
+        data = super()._parse(path)
+        return cls._update_component(data)
+
+    @classmethod
+    def _update_component(cls, data: Dict) -> Dict:
+        component = data.get("component", None)
+
+        if not isinstance(component, Dict):
+            return data
+
+        process_value = component.get("process", None)
+
+        if not isinstance(process_value, str):
+            return data
+
+        if cls._is_valid_process_string(process_value):
+            value_as_int = cls._parse_process(process_value)
+            component.update({"process": value_as_int})
+            data.update({"component": component})
+
+        return data
+
+    @classmethod
+    def _parse_process(cls, values: str) -> int:
+        if ":" in values:
+            semicolon_split_values = values.split(":")
+            start_value = int(semicolon_split_values[0])
+            end_value = int(semicolon_split_values[-1])
+            return end_value - start_value + 1
+
+        return len(values.split())
+
+    @classmethod
+    def _is_valid_process_string(cls, values: str) -> bool:
```

Comment:
Here we just get a single string value right? Not multitple strings?

Reply:
will rename to process_value